core/sql/src/main/java/org/trafodion/sql/HDFSClient.java

```diff
@@ -0,0 +1,319 @@
+      while (true) {
+         bytesRead = trafHdfsRead();
+         if (bytesRead == -1 || bytesRead == 0)
+            return totalBytesRead;
+         totalBytesRead += bytesRead;
+         if (totalBytesRead == bufLen_)
+              return totalBytesRead;
+      }  
+   } 
+
+   boolean hdfsCreate(String fname , boolean compress) throws IOException
+   {
+     if (logger_.isDebugEnabled()) 
+        logger_.debug("HDFSClient.hdfsCreate() - started" );
+      Path filePath = null;
+      if (!compress || (compress && fname.endsWith(".gz")))
+        filePath = new Path(fname);
+      else
+        filePath = new Path(fname + ".gz");
+        
+      FileSystem fs = FileSystem.get(filePath.toUri(),config_);
+      FSDataOutputStream fsOut = fs.create(filePath, true);
+      
+      if (compress) {
+        GzipCodec gzipCodec = (GzipCodec) ReflectionUtils.newInstance( GzipCodec.class, config_);
+        Compressor gzipCompressor = CodecPool.getCompressor(gzipCodec);
+        outStream_= gzipCodec.createOutputStream(fsOut, gzipCompressor);
+      }
+      else
+        outStream_ = fsOut;      
+      if (logger_.isDebugEnabled()) 
+         logger_.debug("HDFSClient.hdfsCreate() - compressed output stream created" );
+      return true;
+    }
+    
+    boolean hdfsWrite(byte[] buff, long len) throws IOException
+    {
+
+      if (logger_.isDebugEnabled()) 
+         logger_.debug("HDFSClient.hdfsWrite() - started" );
+      outStream_.write(buff);
+      outStream_.flush();
+      if (logger_.isDebugEnabled()) logger_.debug("HDFSClient.hdfsWrite() - bytes written and flushed:" + len  );
+      return true;
+    }
+    
+    boolean hdfsClose() throws IOException
+    {
+      if (logger_.isDebugEnabled()) logger_.debug("HDFSClient.hdfsClose() - started" );
+      if (outStream_ != null) {
+          outStream_.close();
+          outStream_ = null;
+      }
+      return true;
+    }
+
+    
+    public boolean hdfsMergeFiles(String srcPathStr, String dstPathStr) throws IOException
+    {
+      if (logger_.isDebugEnabled()) logger_.debug("HDFSClient.hdfsMergeFiles() - start");
+      if (logger_.isDebugEnabled()) logger_.debug("HDFSClient.hdfsMergeFiles() - source Path: " + srcPathStr + 
+                                               ", destination File:" + dstPathStr );
+        Path srcPath = new Path(srcPathStr );
+        srcPath = srcPath.makeQualified(srcPath.toUri(), null);
+        FileSystem srcFs = FileSystem.get(srcPath.toUri(),config_);
+  
+        Path dstPath = new Path(dstPathStr);
+        dstPath = dstPath.makeQualified(dstPath.toUri(), null);
+        FileSystem dstFs = FileSystem.get(dstPath.toUri(),config_);
+        
+        if (dstFs.exists(dstPath))
+        {
+          if (logger_.isDebugEnabled()) logger_.debug("HDFSClient.hdfsMergeFiles() - destination files exists" );
+          // for this prototype we just delete the file-- will change in next code drops
+          dstFs.delete(dstPath, false);
```

Comment:
Maybe this, and the following two lines of comments should just be removed? Are we supposed to silently delete existing file and overwrite with new merged file?